lib/runtime-core/src/vm.rs

```diff
@@ -24,6 +25,8 @@ pub struct Ctx {
 
     pub data: *mut c_void,
     pub data_finalizer: Option<extern "C" fn(data: *mut c_void)>,
+
+    pub maybe_symbol_map: Option<HashMap<u32, String>>,
```

Comment:
I think `symbol_map` would also be an okay name for this, I do like `maybe_symbol_map` as a variable name but it looks a little weird to me as a field name since I don't think we do this anywhere else.